Sources/SATSCore/BasicComponents/CustomAsyncImage/ImageViewModel.swift

```diff
@@ -0,0 +1,28 @@
+import SwiftUI
+
+/// Implementation detail for `CustomAsyncImage`
+class ImageViewModel: ObservableObject {
+    @Published var state: ImageViewData = .empty
+
+    private var imageClient: ImageClient { Config.imageClient }
```

Comment:
This dependency should be injected in the constructor. 

```suggestion
init(
    state: ImageViewData,
    imageClient: ImageClient = Config.imageClient
) {
    self.state = state
    self.imageClient = imageClient
}
```

Reply:
I did this on purpose.

`ImageViewModel` is a internal type, so the consumer of the library cannot initialize/pass the client along.

`CustomAsyncImage` will create instances of `ImageViewModel`, you can argue that you can inject `ImageClient` to `CustomAsyncImage`, but that would result in a bit of a overly complicated API, as you will use this component inside other components and so on.

For that reason I felt it was more appropriate to inject it to this Singleton proxy object, as Image loading in general is quite a common/generic problem.

Reply:
That sounds fine